### DISCUSSION TITLE
Improved instruction on use of Docker

### DISCUSSION THREAD

**Discussion by Waxwood** [TriliumNext/Trilium#1413] [Q&A]
I found difficulty understanding how to get Trilium to use Docker. 
I previously installed Trilium as a local installation using Manjaro Linux "AUR" installation process. 

After reading a great deal about Docker, I installed it on my desktop Manjaro system, and installed Kitematic GUI. 
I can see that "Trilium:latest" image is installed.
It seems that the Trilium image (docker pull zadam/trilium:[VERSION]) runs as a container in Docker on my computer. 

When I run the image using "sudo docker run -t -i -p 127.0.0.1:8080:8080 -v ~/trilium-data:/root/trilium-data zadam/trilium:latest
", Trilium starts with a setup screen. If I chose  "I have desktop instance already and want to syn with it", then the next screen asks me "1. Please open your desktop instance of Trilium Notes; 2. Click on options; Click on Sync etc"  but I cannot do this because I can no longer open my installation of Trilium desktop. Each time I try to reopen, the Trilium Notes startup will begin. How can this be resolved?

Does installing the Trilium Docker in addition to having a local installation mean that I now have two versions of Trilium on my system? Can I delete the local installation? Also, how does the Trilium Docker version sync? What is it syncing to? I originally thought the Docker website offered some type of file hosting, but it appears to be only a source for Docker Images.

If I install Triliium docker version on another my notebook (in addition to the original installation on my desktop), will they sync together?

I'm very new to Dockers and am having a lot of trouble understanding how this works. Probably many others have similar questions. Thank you for clarifying this.

**Comment by Waxwood** [comment]
I've been unable to access my Trilium database since this problem occurred. I hope there's an easy fix :)

**Comment by zadam** [comment]
Hi, I don't fully understand all of what you wrote but some points:

> Does installing the Trilium Docker in addition to having a local installation mean that I now have two versions of Trilium on my system?

Yes.

> Can I delete the local installation?

Sure, depending on what you want to do.

> Also, how does the Trilium Docker version sync? What is it syncing to?

Trilium docker is typically used for server installation which can be used as a synchronization point - i.e. that's where your other (desktop) instances will sync through.

> I originally thought the Docker website offered some type of file hosting, but it appears to be only a source for Docker Images.

No file hosting on docker. Images only.

> If I install Triliium docker version on another my notebook (in addition to the original installation on my desktop), will they sync together?

That's exactly why you need to have that server installation on some public hosting so that your e.g. notebook and desktop installation can sync through it.

**Reply by Waxwood** [reply]
Thank you for your reply. Sorry for my inability to clearly explain my situation- I'm very new to the idea of Dockers, and have no experience with server installation etc.

Essentially, my problem is this:
I installed Trilium on Manjaro linux system using AUR. 
It installed easily and I imported a large database from Joplin into Trilium and continued to expand my use of Trilium.

In my attempt to try to figure out how to sync Trilium on my desktop to Trilium on my notebook, I read about Dockers and installed Trilium on my desktop system. However now, when I click on the Trilium icon, the startup screen begins and provides 3 startup option. I don't want to begin anew database, but when I click on "I have a desktop instance and want to sync with it", the next screen asks me to start my desktop instance. But I don't know how to start it. 

I am guessing that the Trilium installed from Manjaro AUR has the userfile database in a location different from the Docker version. 
Can I move my database from its curent location to the location used by the Docker version of Trilium? Will this Docker version then function as a "server" version to which I can connect/sync my notebook installation of Trlium? 
I'll wait for advice before I try anything. I don't want to accidentally corrupt or delete my database. 
Thanks!!

**Reply by Waxwood** [reply]
I removed Docker and all images assuming the local install version of Trilium would work again. When I start it, the startup screen begins and provides 3 startup options. I reinstalled Docker and the Trilium image but I receive this message: 
```
Generated session secret
DB size: 4 KB
DB not initialized, please visit setup page - http://[your-server-host]:8080 to see instructions on how to initialize Trilium.
App HTTP server starting up at port 8080
{
  "appVersion": "0.45.4",
  "dbVersion": 170,
  "syncVersion": 16,
  "buildDate": "2020-11-12T22:15:23+01:00",
  "buildRevision": "6c57b2220ff05059d7460369b195d281fcd9cbb6",
  "dataDirectory": "/root/trilium-data",
  "clipperProtocolVersion": "1.0"
}
CPU model: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz, logical cores: 8 freq: 4022 Mhz
Listening on port 8080
```

Sadly, I have no idea how to fix it. 
The Trilium Docker Server installation recommends running `sudo docker run -t -i -p 127.0.0.1:8080:8080 -v ~/trilium-data:/root/trilium-data zadam/trilium` to ensure the installation is mounted and persistent. Do I need to run this with SUDO if I have already added my username to the Docker permission group? I tried both ways.
But after running that command in terminal, I receive this message: 
```
[ray1@Giant-Raffy ~]$ docker run -t -i -p 127.0.0.1:8080:8080 -v ~/trilium-data:/root/trilium-data zadam/trilium
docker: Error response from daemon: driver failed programming external connectivity on endpoint nice_jackson (06cf5da48349158fb55252b3976089389e8520982eddb67cb21cd1f98ca241d9): Error starting userland proxy: listen tcp 127.0.0.1:8080: bind: address already in use.
ERRO[0000] error waiting for container: context canceled 
```
I'm at a loss on how to get this up and running. Thank you for advice on how to fix this :)
